clients/feeder/feeder.go

```diff
@@ -73,8 +74,12 @@ func (c *Client) WithUserAgent(ua string) *Client {
 	return c
 }
 
-func (c *Client) WithTimeout(t time.Duration) *Client {
-	c.client.Timeout = t
+func (c *Client) WithTimeouts(timeouts []time.Duration, fixed bool) *Client {
+	if fixed {
+		c.timeouts = getFixedTimeouts(timeouts)
+	} else {
+		c.timeouts = getDynamicTimeouts(timeouts)
+	}
```

Comment:
In relation with a previous comment, regarding `getDynamicTimeout` based on a single duration.

```suggestion
func (c *Client) WithTimeouts(timeouts []time.Duration, fixed bool) *Client {
	if len(timeouts) > 1 || fixed {
		c.timeouts = getFixedTimeouts(timeouts)
	} else {
		c.timeouts = getDynamicTimeouts(timeouts[0])
	}
```